### PR TITLE
INTENG-10910/pickup adobe event names correctly

### PR DESCRIPTION
**Pull request by bklastaitis-branch**
## Reference
https://branch.atlassian.net/browse/INTENG-10910

## Description
pick up event names when they are fired via `MobileCore.trackState` and/or `MobileCore.trackAction`, add better tests

## Testing Instructions
Run tests inside `AdobeEventTest` individually, there are some codependencies between tests that make them fail when they run altogether and I didn't have the time to solve that.

## Risk Assessment [`HIGH | MEDIUM | LOW`]
MEDIUM

- [x] I, the PR creator, have tested — integration, unit, or otherwise — this code.

## Reviewer Checklist (To be checked off by the reviewer only)

**You can only merge once all of these are checked off!**

- [x] JIRA Ticket is referenced in MR title
- Correctness & Style
    - [x] Conforms to [Style Guides](https://source.android.com/setup/contribute/code-style)
    - [x] Mission critical pieces are documented in code and out of code as needed
- [x] Unit Tests reviewed and test issue sufficiently
- [ ] Functionality was reviewed in QA independently by another engineer on the team (clone for each required reviewer)

### CODE REVIEW

**Review by echo-branch — APPROVED** [review]

**Comment by echo-branch** [comment]
Code looks good, but I'd feel better with an end to end test to confirm it looks right on the dashboard.

**Comment by bklastaitis-branch** [comment]
Thanks @echo-branch! I will make sure to test it end-to-end before releasing. I won't merge it until then in case a second review is needed.

**Comment by bklastaitis-branch** [comment]
Tested end to end, was able to see events on both dashboards [Adobe](https://experience.adobe.com/#/@branchmetricsexchange/analytics/spa/index.html?path=launch&defaultLandingPage=true&ssSession=1e9a5a6083907b2edd56d6f29d6d2d76&jpj=54138053804741#/workspace/template/sample) and Branch. Will add release notes, bump up version code, merge and release.

**Pull request merged**
